brayns/utils/image/ImageCodecRegistry.cpp

```diff
@@ -85,8 +88,10 @@ class ImageCodecStorage
 
 namespace brayns
 {
-const ImageCodec &ImageCodecRegistry::getCodec(const std::string &format)
+const ImageCodec &ImageCodecRegistry::getCodec(std::string format)
 {
+    StringCase::lower(format);
```

Comment:
I think this is misleading to make a copy of the string, we just want to read it, not take ownership of it.

You can use StringCase::toLower to make a lower copy of it.

Reply:
I dont see the misleading part of that. Besides, this is not taking ownership of the string. Is making a copy out of it.

Reply:
That's what I say, with the copy in the signature, it looks like we are taking ownership of the string while we just read it (the lower copy is just an implementation detail, it shouldn't go in the function signature (imo).

But I did this before C++17, now the best would be string_view as we don't need specifically an std::string...

Reply:
I still dont see the problem of having the copy in the signature, what is confusing about it?

Reply:
As I said I would leave the copy as an implementation detail so we don't see it in the signature and it is clear the input is just something we are reading from (with const string & or string_view).

For me when I see an parameter expensive to copy passed by value, it means that I am transfering its ownership to the object I call (like with unique_ptr).

But that's just personal feeling, you can leave it like that if you prefer.

Reply:
I believe the correct way is to leave like that, because the end goal is to make a copy of the string, thus the user is aware that will be done. Passing a const std::string & or an std::string_view hides this fact.

Furthermore, I believe your definition of "transfer ownership" is wrong. This would imply that the caller is loosing the object he is passing (thats why the definition works for an unique_ptr, its mean to be owned by in just one place, hence the need to cast it to an rvalue reference if its an lvalue one). In this case, it is a copy, and the caller still owns and can manipulate as he pleases his object.